runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,267 @@
+import net.runelite.client.ui.overlay.worldmap.WorldMapPointManager;
+
+@PluginDescriptor(
+		name = "Death Indicator"
+)
+@Slf4j
+public class DeathIndicatorPlugin extends Plugin
+{
+	public static final BufferedImage WORLDMAP_ARROW;
+	public static final BufferedImage BONES;
+
+	private boolean worldMapPointSet = false;
+	private boolean hasDied = false;
+	private Timer deathTimer = null;
+	private static int currentWorld = -1;
+	private static int deathWorld = -1;
+	private WorldPoint deathLocation = null;
+	private WorldPoint currentLocation = null;
+	private boolean deathMarkerReset = false;
+	private boolean playerHasRespawned = true;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private DeathIndicatorConfig config;
+
+	@Inject
+	private DeathIndicatorPlugin plugin;
+
+	@Inject
+	private WorldMapPointManager worldMapPointManager;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	static
+	{
+		try
+		{
+			synchronized (ImageIO.class)
+			{
+				WORLDMAP_ARROW = ImageIO.read(DeathIndicatorPlugin.class.getResourceAsStream("worldmap_arrow.png"));
+				BONES = ImageIO.read(DeathIndicatorPlugin.class.getResourceAsStream("bones.png"));
+			}
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Provides
+	DeathIndicatorConfig deathIndicatorConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DeathIndicatorConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		client.clearHintArrow();
+
+		if (worldMapPointSet)
+		{
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			worldMapPointSet = false;
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onActorDeath(ActorDeath event)
+	{
+		if (client.isInInstancedRegion())
+		{
+			return;
+		}
+
+		Actor actor = event.getActor();
+
+		if (actor instanceof NPC || actor != client.getLocalPlayer())
+		{
+			return;
+		}
+
+		hasDied = true;
+		playerHasRespawned = false;
+
+		deathLocation = actor.getWorldLocation();
+		deathWorld = client.getWorld();
+
+		if (config.showDeathHintArrow())
+		{
+			client.setHintArrow(deathLocation);
+		}
+
+		if (config.showDeathOnWorldMap())
+		{
+			addMapPoint(deathLocation);
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+
+		if (config.showDeathInfoBox())
+		{
+			deathTimer = new Timer(60, ChronoUnit.MINUTES, BONES, plugin);
+			deathTimer.setTooltip("Died on world: " + deathWorld);
+			infoBoxManager.addInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getMessage().equals("Oh dear, you are dead!"))
+		{
+			playerHasRespawned = true;
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!hasDied)
+		{
+			return;
+		}
+
+		if (!playerHasRespawned)
+		{
+			return;
+		}
```

Comment:
This should just be moved to the statement above

Reply:
I'm not exactly sure why, but when I do combine these two statements the entire plugin breaks. I think it's because they fire on game tick events and the death occurs 5 seconds before the respawn occurs so if they are together I miss the opportunity to move forward and document the death.